src/job_cache/message_parser.h

```diff
@@ -55,13 +67,19 @@ struct MessageParser {
         // as equivlient to a close which would normally appear as the count == 0
         // case above.
         if (errno == ECONNRESET) return MessageParserState::StopSuccess;
-        // There are some failures that could occur that just require us to retry.
-        // EINTR could occur on any fd type but EAGAIN and EWOULDBLOCK should
-        // only occur if the user gave us a non-blocking socket.
-        if (errno == EINTR || errno == EAGAIN || errno == EWOULDBLOCK) continue;
+        // On EINTR we should just retry until we get EAGAIN/EWOULDBLOCK
+        if (errno == EINTR) {
+          continue;
+        }
+        // If we hit EAGAIN/EWOULDBLOCK then we might have more work to do but
+        // we can't do that work just yet and need to continue.
+        if (errno == EAGAIN || errno == EWOULDBLOCK)
+            return MessageParserState::Continue;
```

Comment:
I personally  prefer wrapping in  `{}`  but at  the very least the brackets should be consistent. Right now this function has three  different styles

Reply:
Yeah I'll add the brackets